core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -260,7 +315,23 @@ private MethodMetrics getMethodMetrics(RequestEvent event) {
       if (method == null) {
         return null;
       }
-      return this.metrics.get(method.getInvocable().getDefinitionMethod());
+
+      RuntimeMetrics metrics = this.metrics.get(method.getInvocable().getDefinitionMethod());
+      if (metrics == null) {
+        return null;
+      }
+
+      Map tags = null;
+      if (event.getContainerRequest() != null) {
+        tags = (Map) event.getContainerRequest().getProperty(RUNTIME_TAGS_PROP_KEY);
+      }
+
+      if (tags == null) {
+        return metrics.metrics();
+      }
+
+      // we have additional tags, find the appropriate metrics holder
+      return metrics.metrics(tags);
```

Comment:
can we just use the same metrics method for both scenarios -no additional tags and additional tags

Reply:
Metrics were  immutable in this project (since the underlying sensor objects are also immutable). Once created they hold their tags for the application lifetime.

Reply:
oh, I meant instead of two methods metrics() and metrics(tags) can we just use metrics(tags) ?

Reply:
I think it would be ok to keep two different methods. would it clearer if we renamed `metrics(tags)` to something like `requestMetrics(tags)`?